### PR TITLE
Strictly monotonic

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/pandas-dev/pandas/issues/16515

I think this can go in 0.20.2, even though part of it is a (small) enhancement.

### CODE REVIEW

File: pandas/tests/indexes/test_multi.py [hunk 1]
```diff
@@ -2416,6 +2426,20 @@ def test_is_monotonic(self):
...
         assert not i.is_monotonic
+        assert not i.is_strictly_monotonic_increasing
+
+    def test_is_strictly_monotonic(self):
+        idx = pd.MultiIndex(levels=[['bar', 'baz'], ['mom', 'next']],
+                            labels=[[0, 0, 1, 1], [0, 0, 0, 1]])
+        assert idx.is_monotonic_increasing
+        assert not idx.is_strictly_monotonic_increasing
+
+    @pytest.mark.xfail(reason="buggy MultiIndex.is_monotonic_decresaing.")
```
Comment (TomAugspurger) [inline pandas/tests/indexes/test_multi.py]: https://github.com/pandas-dev/pandas/issues/16554

**Comment by jorisvandenbossche** [comment]
Would dask like to use those `is_strictly_monotonic_increasing` ? Otherwise keeping them private to not further increase number of methods is also an option.

**Comment by jreback** [comment]
lgtm. merge away.

**Comment by jreback** [comment]
thanks!

**Comment by jorisvandenbossche** [comment]
Asking again, as you merged without responding to my comment: is it needed to add this to the public API? It there demand for this attribute? 
If there is not an actual usecase (apart from the internal usage to fix the bug), I would consider making them not public

**Comment by TomAugspurger** [comment]
I'm slightly in favor of having it public. In terms of cognitive load, this feels like half an additional method, rather than 2 additional methods 😄  Just since these are so similar to `is_monotonic_increasing` and decreasing in name and functionality.

But happy hide it if you want.

**Comment by jorisvandenbossche** [comment]
> In terms of cognitive load, this feels like half an additional method, rather than 2 additional methods :smile: Just since these are so similar to is_monotonic_increasing

Yes, but in tab completion they don't appear together, so I would say it is back to a full additional method in feeling :-)

Since it is very easy to do this yourself as you said above (just combine it with `is_unique`), I would rather add that as an example to the docs and leave them private

**Comment by TomAugspurger** [comment]
> Since it is very easy to do this yourself as you said above (just combine it with is_unique), I would rather add that as an example to the docs and leave them private
 
Perhaps I'm slow, but it took me a day to realize that strictly monotonic = monotonic & unique 😆 

PR is at https://github.com/pandas-dev/pandas/pull/16576 Let's hide it for now, and if anyone asks for it we can make it public.

**Comment by jreback** [comment]
@jorisvandenbossche I thought this was fine. yes its a a convience method. Sure you *can* do it, but why not. ok with re-hiding it as @TomAugspurger PR.

**Pull request merged**